packages/client/hmi-client/src/components/responsive-matrix/cell-bar-graph.vue

```diff
@@ -0,0 +1,282 @@
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	data() {
+		return {
+			topMargin: 10,
+			bottomMargin: 20,
+			leftMargin: 30,
+			rightMargin: 20,
+			containerBoundingBox: {} as DOMRect,
+			svg: null as any
+		};
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// computed                                                                     //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	computed: {
+		selectedCells(): any {
+			const startRow = this.selectedCell[SelectedCellValue.START_ROW];
+			const endRow = this.selectedCell[SelectedCellValue.END_ROW];
+			const startCol = this.selectedCell[SelectedCellValue.START_COL];
+			const endCol = this.selectedCell[SelectedCellValue.END_COL];
+
+			// const selectedCells: any[] = [];
+			const selectedCells = this.parameters.reduce((acc, param) => {
+				acc[param] = [];
+				return acc;
+			}, {});
+
+			for (let row = startRow, i = 0; row <= endRow; row++, i++) {
+				this.dataRowList[row].forEach((cell: any) => {
+					if (cell.col <= endCol && cell.col >= startCol) {
+						this.parameters.forEach((param) => selectedCells[param].push(cell[param]));
+					}
+				});
+			}
+
+			return selectedCells;
+		},
+
+		labelRowSelected(): number[] {
+			const startRow = this.selectedCell[SelectedCellValue.START_ROW];
+			const endRow = this.selectedCell[SelectedCellValue.END_ROW];
+			return this.labelRowList.slice(startRow, endRow + 1);
+		},
+
+		labelRowSelectedIndexMap(): number[] {
+			return this.labelRowSelected;
+		},
+
+		labelColSelected(): number[] {
+			const startCol = this.selectedCell[SelectedCellValue.START_COL];
+			const endCol = this.selectedCell[SelectedCellValue.END_COL];
+			return this.labelColList.slice(startCol, endCol + 1);
+		},
+
+		// assume that timestep for all cells in same column are equal
+		labelColSelectedMin(): number {
+			return Math.min(...this.labelColSelected);
+		},
+
+		// assume that timestep for all cells in same column are equal
+		labelColSelectedMax(): number {
+			return Math.max(...this.labelColSelected);
+		},
+
+		parametersMinAll(): number {
+			return Math.min(...(Object.values(this.parametersMin) as number[]));
+		},
+
+		parametersMaxAll(): number {
+			return Math.max(...(Object.values(this.parametersMax) as number[]));
+		},
+
+		xRange(): any {
+			return [0, this.containerBoundingBox.width - this.leftMargin - this.rightMargin];
+		},
+
+		xScaleLinear(): any {
+			return scaleLinear()
+				.domain([this.labelColSelectedMin, this.labelColSelectedMax])
+				.range(this.xRange);
+		},
+
+		xScaleBand(): any {
+			return scaleBand().range(this.xRange).domain(this.parameters).padding(0.15);
+		},
```

Comment:
Some of these compute values are not being used, like `xScaleLinear` and `xScaleBand`  - if you want to keep them around for reference let's comment them out, if not let's remove them.

Reply:
ya we can remove them